bindings/c/include/khiva_c/defines.h

```diff
@@ -7,9 +7,21 @@
 #if defined(_WIN32) || defined(_MSC_VER)
 // http://msdn.microsoft.com/en-us/library/b0084kay(v=VS.80).aspx
 // http://msdn.microsoft.com/en-us/library/3y1sfaz2%28v=VS.80%29.aspx
-#define KHIVAAPI __declspec(dllexport)
+#define DLL_EXPORT __declspec(dllexport) // Note: actually gcc seems to also supports this syntax.
+#define DLL_IMPORT __declspec(dllimport)
 #else
+#define DLL_EXPORT
+#define DLL_IMPORT
+#endif
+
+#ifdef KHIVA_STATICLIB
 #define KHIVAAPI
+#else
+#if defined(BUILDING_KHIVA_DLL)
+#  define KHIVAAPI DLL_EXPORT
```

Comment:
please take a look at the definition of KHIVAAPI. It was failing on Linux/Mac

Reply:
I think I already fixed that on the last commit https://github.com/shapelets/khiva/pull/15/commits/459e6dede54de8674b6fd51efa3f56b3d9b972ea

Reply:
ok. I have a question here, KHIVAAPI is a define used in the bindings to export the functions. Shouldn't that be always an export? I think we should use the dllimport to import core functions to use them in the bindings. And this in the case we compile it as a dynamic library. Right?

Reply:
When you build the Khiva C lib, KHIVAAPI has to expand to an export declaration to have the symbols exported. But when you use Khiva C headers in the library that uses the Khiva C lib, it has to expand to an import declaration otherwise it won't find the symbols and it will probably lead to linker errors. That is why the Khiva C lib has the BUILDING_KHIVA_DLL added as compile definition, to expand KHIVAAPI to a __declspec(dllexport). Otherwise __declspec(dllimport) is defined, which is the case of the libraries that use Khiva C lib.

Reply:
Lol, I just realized that the definition of BUILDING_KHIVA_DLL has to be PRIVATE, to avoid being transitively passed to the users of the lib.

Reply:
Yes, this is only applicable to shared libraries. But khiva C is forced to be built shared. Another approach would be to use the CMake variable BUILD_SHARED_LIBS and set it to on or off and adding your libraries without any STATIC nor SHARED modifiers and let the user choose. In the latter case you should take into account the value of BUILD_SHARED_LIBS in order to add the correct compile definitions to export or import the symbols.

Reply:
Ok gotcha